### PR TITLE
Clarify packages in overview

### PR DESCRIPTION
**Pull request by Frozzzie**
A repeated occurrence in the #beginner channel of the Odin discord chat is new users being confused about how packages work. This is a clarification intended to answer those questions at the outset. 

Language is intended to stay neutral and simply present how things work, in line with the rest of the overview. Please review and consider merging.

### CODE REVIEW

File: content/docs/overview.md [hunk 1]
```diff
@@ -160,7 +160,7 @@ For more information regarding value declarations in general, please see the [Od
 
 
 ## Packages
-Every Odin program is made up of packages. Programs begin running in the package `main`.
+Every Odin program is made up of packages. A package is a directory of Odin code files, with each file in that directory having the same package declaration at the top. Programs begin running in the package `main`.
```
Comment (Kelimion) [inline content/docs/overview.md:163]: I'd condense this a bit:

Odin programs consist of packages. A package is a directory of Odin code files, all of which have the same package declaration at the top. Execution starts in the package's `main` procedure.

File: content/docs/overview.md [hunk 2]
```diff
@@ -204,6 +204,22 @@ my_variable: int // cannot be accessed outside this file.
 ```
 `@(private)` is equivalent to `@(private="package")`.
 
+### Authoring a package
+
+A package is a directory of Odin code files which are marked with the same package header: `package main`
```
Comment (Kelimion) [inline content/docs/overview.md:209]: And here I'd repeat the same phrasing:

A package is a directory of Odin code files, all of which have the same package declaration at the top, e.g. `package main`. Each `.odin` file _must_ have the same package name. A directory cannot contain more than 1 package.

File: content/docs/overview.md [hunk 3]
```diff
@@ -204,6 +204,22 @@ my_variable: int // cannot be accessed outside this file.
...
+Each file in a directory _must_ have the same package declaration. A directory cannot contain multiple packages. 
+
+### Subpackages
```
Comment (Kelimion) [inline content/docs/overview.md:212]: I would call this `### Organizing packages`. They're not technically sub-packages.

They're not properly sub-packages, even though it's easy to think of them as such.

Perhaps:
```
### Organizing packages

Packages may be thematically organized by placing them in subdirectories of another package. For example: `core:image/png` and `core:image/tga`, as subdirectories of `core:image`. 
Nesting these packages is a helpful taxonomy. It does not imply a dependency: `core:foo/bar` does not need to import `core:foo` and reference anything from it.


```

**Review by Kelimion — CHANGES_REQUESTED** [review]

**Comment by Skytrias** [comment]
good work 👍 
i wonder whether we should start taking things out of the overview and push new "posts" which can focus on topics like packages more

File: content/docs/overview.md (hunk 3 above)
Comment (Frozzzie) [reply content/docs/overview.md:212]: Agreed, I think having the overview be clear, factual statements is the way to go. I'll update.

File: content/docs/overview.md (hunk 1 above)
Comment (Frozzzie) [reply content/docs/overview.md]: Taken this wording as is. Thank you.

File: content/docs/overview.md (hunk 2 above)
Comment (Frozzzie) [reply content/docs/overview.md]: As above.

**Comment by Kelimion** [comment]
This was a good idea. Thanks for the collab, Barker.

**Pull request merged**